testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java

```diff
@@ -508,4 +508,11 @@ public void testUnionAllAboveBroadcastJoin()
     {
         assertQuery("SELECT COUNT(*) FROM region r JOIN (SELECT nationkey FROM nation UNION ALL SELECT nationkey as key FROM nation) n ON r.regionkey = n.nationkey", "VALUES 10");
     }
+
+    @Test
+    public void testDistinctAggregationWithConstantLiterals()
+    {
+        String query = "select count(distinct 'abc'), count(*) from orders";
```

Comment:
the `count(distinct 'abc')` part might be optimized away in future maybe. Seems like a brittle test. Why not apply `count(distinct)` on a column that actually exists?

Reply:
The issue happens when we apply a count distinct of a constant literal - if we apply `count(distinct)` on a column which exists `SimplifyCountOverConstant` doesn't modify the underlying plan.